packages/core/src/modules/proofs/ProofsModule.ts

```diff
@@ -145,14 +154,59 @@ export class ProofsModule {
       requestedPredicates: proofRequestOptions.requestedPredicates,
     })
 
-    const { message, proofRecord } = await this.proofService.createRequest(connection, proofRequest, config)
+    const { message, proofRecord } = await this.proofService.createRequest(
+      { proofRequest, comment: proofRequestOptions.comment },
+      connection
+    )
 
     const outboundMessage = createOutboundMessage(connection, message)
     await this.messageSender.sendMessage(outboundMessage)
 
     return proofRecord
   }
 
+  /**
+   * Initiate a new presentation exchange as verifier by creating a presentation request
+   * not bound to any connection. The request must be delivered out-of-band to the holder
+   *
+   * @param proofRequestOptions Options to build the proof request
+   * @returns The proof record and proof request message
+   *
+   */
+  public async createOutOfBandRequest(proofRequestOptions: ProofRequestOptions): Promise<{
+    requestMessage: RequestPresentationMessage
+    proofRecord: ProofRecord
+  }> {
+    const nonce = proofRequestOptions.nonce ?? (await this.proofService.generateProofRequestNonce())
+
+    const proofRequest = new ProofRequest({
+      name: proofRequestOptions.name ?? 'proof-request',
+      version: proofRequestOptions.name ?? '1.0',
+      nonce,
+      requestedAttributes: proofRequestOptions.requestedAttributes,
+      requestedPredicates: proofRequestOptions.requestedPredicates,
+    })
+
+    const { message, proofRecord } = await this.proofService.createRequest({
+      proofRequest,
+      comment: proofRequestOptions.comment,
+    })
+
+    // Create and set ~service decorator
+    const routing = await this.mediationRecipientService.getRouting()
+    message.service = new ServiceDecorator({
+      serviceEndpoint: routing.endpoint,
+      recipientKeys: [routing.verkey],
+      routingKeys: routing.routingKeys,
+    })
+
+    // Save ~service decorator to record (to remember our verkey)
+    proofRecord.requestMessage = message
+    await this.proofRepository.update(proofRecord)
+
+    return { proofRecord, requestMessage: message }
+  }
+
   /**
    * Accept a presentation request as prover (by sending a presentation message) to the connection
    * associated with the proof record.
```

Comment:
I built JS doc blindness so I usually don't read it so much :) but I'm not sure if it's updated according to the changes.